src/concrete/ml/sklearn/base.py

```diff
@@ -651,41 +799,59 @@ def predict(self, X: Data, fhe: Union[FheMode, str] = FheMode.DISABLE) -> numpy.
         # Quantize the input
         q_X = self.quantize_input(X)
 
+        predict_method: Callable
         # If the inference is executed in FHE or simulation mode
         if fhe in ["simulate", "execute"]:
             # Check that the model is properly compiled
             self.check_model_is_compiled()
 
-            q_y_pred_list = []
-            for q_X_i in q_X:
-                # Expected encrypt_run_decrypt output shape is (1, n_features) while q_X_i
-                # is of shape (n_features,)
-                q_X_i = numpy.expand_dims(q_X_i, 0)
```

Comment:
wasn't this necessary?

Reply:
it's just been moved a bit lower in the file